productivity-suite/piercing-library/src/piercing-gateway.ts

```diff
@@ -135,30 +144,33 @@ export class PiercingGateway<Env> {
       ?.includes("text/html");
 
     if (requestIsForHtml) {
-      const baseUrl = this.config.getBaseAppUrl(env).replace(/\/$/, "");
+      const baseUrl = this.config.getLegacyAppBaseUrl(env).replace(/\/$/, "");
       const indexBodyResponse = this.fetchBaseIndexHtml(
         new Request(baseUrl, request)
       ).then((response) => response.text());
 
+      const piercingEnabled =
+        !this.config.shouldPiercingBeEnabled ||
+        (await this.config.shouldPiercingBeEnabled(request, env, ctx));
+
```

Comment:
Given that we are using `!piercingEnabled` below how about we just do:

```suggestion
      const piercingDisabled = await this.config.shouldPiercingBeEnabled.?(request, env, ctx);
```

Reply:
is this not missing a negation?

if `this.config.shouldPiercingBeEnabled(request, env, ctx)` returns `true`, we get `piercingDisabled === true` which is the opposite of what we want

no?

Reply:
Doh! Yeah sorry. Ignore this.